php/libraries/IdentifierGenerator.php

```diff
@@ -28,13 +28,59 @@ abstract class IdentifierGenerator
 {
     private const RANGE_FULL = 'Cannot create new identifier because all ' .
         ' valid identifiers are in use!';
+    /**
+     * How the ID will be generated. 'sequential' or 'random'.
+     *
+     * @var string
+     */
     protected $generationMethod;
+    /**
+     * An array of all the valid characters that can be used in the ID.
+     *
+     * @var array<int,float|int> This is derived from the return type of PHP's
```

Comment:
Are you sure `$alphabet` will either contain int or float values? I would have guest `char` would be more likely

Reply:
I'll look into it. It's possible that PHP stores this internally as ASCII because i remember phpstan giving an error about this.